qiskit/visualization/pulse_v2/core.py

```diff
@@ -0,0 +1,493 @@
+        snapshot_sched = program.filter(instruction_types=[pulse.instructions.RelativeBarrier])
+        for t0, inst in snapshot_sched.instructions:
+            inst_data = types.NonPulseTuple(t0, self.dt, inst)
+            for gen in PULSE_STYLE['generator.barrier']:
+                for drawing in gen(inst_data):
+                    self._replace_drawing(drawing)
+
+    def set_time_range(self,
+                       t_start: Union[int, float],
+                       t_end: Union[int, float]):
+        """Set time range to draw.
+
+        The update to time range is applied after py:method`update_channel_property` is called.
+
+        Args:
+            t_start: Left boundary of drawing in units of cycle time or real time.
+            t_end: Right boundary of drawing in units of cycle time or real time.
+
+        Raises:
+            VisualizationError: When times are given in float without specifying dt.
+        """
+        # convert into nearest cycle time
+        if isinstance(t_start, float):
+            if self.dt is not None:
+                t_start = int(t_start / self.dt)
+            else:
+                raise VisualizationError('Floating valued start time %f seems to be in '
+                                         'units of sec but dt is not specified.' % t_start)
+        # convert into nearest cycle time
+        if isinstance(t_end, float):
+            if self.dt is not None:
+                t_end = int(t_end / self.dt)
+            else:
+                raise VisualizationError('Floating valued end time %f seems to be in '
+                                         'units of sec but dt is not specified.' % t_end)
+
+        duration = t_end - t_start
+
+        self.bbox_left = t_start - int(duration * PULSE_STYLE['formatter.margin.left'])
+        self.bbox_right = t_end + int(duration * PULSE_STYLE['formatter.margin.right'])
+
+    def update_channel_property(self,
+                                visible_channels: Optional[List[pulse.channels.Channel]] = None,
+                                scales: Optional[Dict[pulse.channels.Channel, float]] = None):
+        """Update channel properties.
+
+        This function updates channels to show and the scaling factor of each channel.
+        Drawing objects generated by `generator.channel` is regenerated and replaced
+        according to the input channel preferences.
+        The `visible`, `scale` and `offset` attribute of each drawing object is also updated.
+
+        This function enables a plotter to dynamically update appearance of output image.
+
+        Args:
+            visible_channels: List of channels to show.
+            scales: Dictionary of scaling factor of channels.
+        """
+        scales = scales or dict()
+
+        # arrange channels to show
+        self.active_channels = self._ordered_channels(visible_channels)
+
+        # new properties
+        chan_visible = {chan: False for chan in self.channels}
```

Comment:
If these are common properties for a channel perhaps they should be stored per-channel in a NamedTuple?